src/itype.h

```diff
@@ -202,29 +202,31 @@ struct islot_brewable {
     time_duration time = 0_turns;
 };
 
-struct islot_armor {
-    /**
-     * Bitfield of enum body_part
-     * TODO: document me.
-     */
-    body_part_set covers;
-    /**
-     * Whether this item can be worn on either side of the body
-     */
-    bool sided = false;
-    /**
-     * How much this item encumbers the player.
-     */
+struct armor_portion_data {
+
+    // How much this piece encumbers the player.
     int encumber = 0;
-    /**
-    * When storage is full, how much it encumbers the player.
-    */
+
+    // When storage is full, how much it encumbers the player.
     int max_encumber = 0;
-    /**
-     * Percentage of the body part area that this item covers.
-     * This determines how likely it is to hit the item instead of the player.
-     */
+
+    // Percentage of the body part that this item covers.
+    // This determines how likely it is to hit the item instead of the player.
     int coverage = 0;
+
+    // Where does this cover if any
+    cata::optional<body_part_set> covers;
+
+    // What layer does it cover if any
+    // TODO: Not currently supported, we still use flags for this
+    cata::optional<layer_level> layer;
```

Comment:
If I understand correctly, you're never assigning to this?  Better  to just remove it than leave it here unused, I think.